google/cloud/bigquerycontrol/integration_tests/job_integration_test.cc

```diff
@@ -0,0 +1,134 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/bigquerycontrol/v2/job_client.h"
+#include "google/cloud/internal/getenv.h"
+#include "google/cloud/testing_util/integration_test.h"
+#include "google/cloud/testing_util/status_matchers.h"
+#include <gmock/gmock.h>
+#include <chrono>
+#include <thread>
+
+namespace google {
+namespace cloud {
+namespace bigquerycontrol_v2 {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+namespace {
+
+using ::google::cloud::testing_util::StatusIs;
+using ::testing::AnyOf;
+using ::testing::Contains;
+using ::testing::Eq;
+using ::testing::HasSubstr;
+using ::testing::IsEmpty;
+using ::testing::Not;
+
+class BigQueryJobIntegrationTest
+    : public ::google::cloud::testing_util::IntegrationTest {
+ protected:
+  void SetUp() override {
+    project_id_ =
+        google::cloud::internal::GetEnv("GOOGLE_CLOUD_PROJECT").value_or("");
+    ASSERT_FALSE(project_id_.empty());
+  }
+  std::string project_id_;
+};
+
+TEST_F(BigQueryJobIntegrationTest, JobCRUD) {
+  namespace bigquery_proto = google::cloud::bigquery::v2;
+  auto client = JobServiceClient(MakeJobServiceConnectionRest());
+
+  bigquery_proto::JobConfigurationQuery query;
+  query.set_query(
+      "SELECT name, state, year, sum(number) as total "
+      "FROM `bigquery-public-data.usa_names.usa_1910_2013` "
+      "WHERE year >= @minimum_year "
+      "GROUP BY name, state, year");
+
+  // Use GoogleSQL dialect to enable parameterized queries.
+  auto use_legacy_sql = google::protobuf::BoolValue();
+  use_legacy_sql.set_value(false);
+  *query.mutable_use_legacy_sql() = use_legacy_sql;
+
+  // Specify value for named integer parameter: @minimum_year
+  query.set_parameter_mode("NAMED");
+  bigquery_proto::QueryParameter minimum_year_param;
+  minimum_year_param.set_name("minimum_year");
+  bigquery_proto::QueryParameterType minimum_year_param_type;
+  minimum_year_param_type.set_type("INT64");
+  *minimum_year_param.mutable_parameter_type() = minimum_year_param_type;
+  bigquery_proto::QueryParameterValue minimum_year_param_value;
+  auto year_value = google::protobuf::StringValue();
+  year_value.set_value("1970");
+  *minimum_year_param_value.mutable_value() = year_value;
+  *minimum_year_param.mutable_parameter_value() = minimum_year_param_value;
+  *query.add_query_parameters() = minimum_year_param;
```

Comment:
Please use `google::protobuf::TextFormat::ParseFromString()`.  In a test, that is a more readable way to initialize complex protos with hard-coded values.